### PR TITLE
chore(docs): Migrate from Docusaurus v2 to v3

### PR DESCRIPTION
**Pull request by jfschubert**
Migrate Docs to latest version of Docusaursu (3.3.2 as of time of writing)

### CODE REVIEW

**Comment by fforootd** [comment]
@hifabienne I finalized the docusaurus 3 migration

Can you review this pr?

In case you are wondering about the changes... https://docusaurus.io/docs/migration/v3 I needed to escape many urls and other paratmeters.

File: docs/docs/guides/integrate/token-introspection/basic-auth.mdx [hunk 1]
```diff
@@ -78,7 +78,7 @@ The request from the API to the introspection endpoint should be in the followin
 
 ```bash
 curl --request POST \
- --url {your_domain}/oauth/v2/introspect \
+ --url \{your_domain}/oauth/v2/introspect \
```
[the remark this answers is not in the document]
Comment (fforootd) [reply docs/docs/guides/integrate/token-introspection/basic-auth.mdx:81]: I think this needs escaping

File: docs/docs/guides/integrate/token-introspection/private-key-jwt.mdx [hunk 2]
```diff
@@ -146,7 +146,7 @@ Create the JSON Web Token with the above header and payload, and sign it with th
 The request from the API to the introspection endpoint should be in the following format:
 ```bash
 curl --request POST \
- --url {your_domain}/oauth/v2/introspect \
+ --url \{your_domain}/oauth/v2/introspect \
```
[the remark this answers is not in the document]
Comment (fforootd) [reply docs/docs/guides/integrate/token-introspection/private-key-jwt.mdx:149]: I think this needs escaping

File: docs/docs/apis/openidoauth/endpoints.mdx [hunk 3]
```diff
@@ -261,7 +261,7 @@ Send a client assertion as JWT for us to validate the signature against the regi
 
 ```BASH
 curl --request POST \
-  --url {your_domain}/oauth/v2/token \
+  --url \{your_domain}/oauth/v2/token \
```
[the remark this answers is not in the document]
Comment (fforootd) [reply docs/docs/apis/openidoauth/endpoints.mdx:264]: https://docusaurus.io/docs/migration/v3#bad-usage-of-

File: docs/docs/guides/integrate/token-introspection/private-key-jwt.mdx (hunk 2 above)
[the remark this answers is not in the document]
Comment (hifabienne) [reply docs/docs/guides/integrate/token-introspection/private-key-jwt.mdx:149]: This is how it looks:
image

[the remark this answers is not in the document]
Comment (fforootd) [reply docs/docs/guides/integrate/token-introspection/private-key-jwt.mdx:149]: hm could be that it works in code snippets already. I will accept these then

**Comment by fforootd** [comment]
@hifabienne I accepted all your changes. Thank you

**Comment by hifabienne** [comment]
@fforootd should we merge this?

**Comment by fforootd** [comment]
> @fforootd should we merge this?

Yes

**Pull request merged**